docs/spark/how-to-guides/streaming.md

```diff
@@ -0,0 +1,123 @@
+
+To start a new connection, open a command prompt. If you are using Linux, run ```nc -lk 9999``` to connect to localhost on port 9999.
+
+If you are using Windows, run ```nc -vvv -l -p 9999``` to connect to localhost port 9999.
+
+Your Spark program listens for the input you type into this command prompt.
+
+### Connect to a stream with ReadStream()
+
+The `ReadStream()` method returns a `DataStreamReader` that can be used to read streaming data in as a `DataFrame`. Include the host and port information to tell your Spark app where to expect its streaming data.
+
+```CSharp
+DataFrame lines = spark
+    .ReadStream()
+    .Format("socket")
+    .Option("host", hostname)
+    .Option("port", port)
+    .Load();
+```
+
+## Register a user-defined function
+
+A UDF is a *user-defined function.* You can use UDFs in Spark applications to perform calculations and analysis on your data.
+
+```CSharp
+Func<Column, Column> udfArray =
+    Udf<string, string[]>((str) => new string[] { str, $"{str} {str.Length}");
+```
+
+In the above code snippet from [StructuredNetworkCharacterCount.cs](StructuredNetworkCharacterCount.cs), you register a UDF called `udfArray`. This UDF processes each string it receives from the netcat terminal to produce an array that includes: the original string (contained in *str*), the original string concatenated with the length of that original string. 
+
+For example, entering *Hello world* in the terminal would produce an array where:
+
+* array[0] = Hello world
+* array[1] = Hello world 11
+
+## Use SparkSQL
+
+Use SparkSQL to perform various functions on the data stored in your DataFrame. It's common to combine UDFs and SparkSQL to apply a UDF to each row of your DataFrame.
+
+```CSharp
+DataFrame arrayDF = lines.Select(Explode(udfArray(lines["value"])));
+```
+
+In the above code snippet from [StructuredNetworkCharacterCount.cs](StructuredNetworkCharacterCount.cs), *udfArray* is applied to each value in your DataFrame (which represents each string read in from our netcat terminal). Apply the SparkSQL method `Explode` to put each entry of your array in its own row. Finally, use `Select` to place the columns you've produced in the new DataFrame *arrayDF.*
+
+## Display your stream
+
+Use `DataFrame.WriteStream()` to establish characteristics of your output, such as printing results to the console, displaying only the most recent output.
+
+```CSharp
+StreamingQuery query = arrayDf
+    .WriteStream()
+    .Format("console")
+    .Start();
+```
+
+## Run your code
+
+Structured streaming in Spark processes data through a series of small **batches.** 
+When you run your program, the command prompt where you establish the netcat allows you to start typing.
+
+When you hit *enter* after entering data in the command prompt, Spark considers that a batch and runs the UDF. 
+
+### Run on Windows
```

Comment:
Users can run on any platform (not just Windows), it's just that the code example for running that we have is designed for Windows. Perhaps change this title so that users don't think they have to run using Windows.

```suggestion
### Use spark-submit to run your app
```